core/src/Platforms/netcore/NetCorePlatformProxy.cs

```diff
@@ -108,6 +109,21 @@ public string GetProductName()
             return _isMsal ? "MSAL.NetCore" : null;
         }
 
+        public string GetApplicationName()
+        {
+            return Assembly.GetEntryAssembly().GetName().Name.ToString();
+        }
+
+        public string GetApplicationVersion()
+        {
+            return Assembly.GetEntryAssembly().GetName().Version.ToString();
+        }
+
+        public string GetDeviceId()
+        {
+            return null;
```

Comment:
Do you want to use Enviroment.MachineName or a hash of that ?

Reply:
done

Reply:
pls. ensure to hash it...

---
In reply to: [228235975](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1352#discussion_r228235975) [](ancestors = 228235975)

Reply:
Hashed in DefaultEvent